feat: 添加组件无障碍功能支持

yangyanAurora opened the pull request:
- button
- Checkbox
- Countdown
- Empty
- Image
- InputNumber
- Price
- Progress
- NavBar
- NoticeBar
- Radio
- Switch
- Tabbar
- Tabs
- Toast
- Popup
- Dialog
<!--
非常感谢您的贡献 维护者审核通过后会合并。
请确保填写以下 pull request 的信息，谢谢！~
-->

### 🤔 这个变动的性质是？

- [x] 新特性提交
- [ ] 日常 bug 修复
- [ ] 站点、文档改进
- [ ] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] TypeScript 定义更新
- [ ] 包体积优化
- [ ] 性能优化
- [ ] 功能增强
- [ ] 国际化改进
- [ ] 重构
- [ ] 代码风格优化
- [ ] 测试用例
- [ ] 分支合并
- [ ] 其他改动（是关于什么的改动？）

### 🔗 相关 Issue

<!--
1. 描述相关需求的来源，如相关的 issue 讨论链接。
-->

### 💡 需求背景和解决方案

<!--
1. 要解决的具体问题。
2. 列出最终的 API 实现和用法。
3. 涉及UI/交互变动需要有截图或 GIF。
-->

### ☑️ 请求合并前的自查清单

⚠️ 请自检并全部**勾选全部选项**。⚠️

- [ ] 文档已补充或无须补充
- [ ] 代码演示已提供或无须提供
- [ ] TypeScript 定义已补充或无须补充
- [ ] fork仓库代码是否为最新避免文件冲突
- [ ] Files changed 没有 package.json lock 等无关文件


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## 发布说明

* **新功能**
  * 为倒计时、图片、输入数字、导航栏、通知栏、价格、进度条、轮播、标签栏、标签面板、对话框等多个组件增强了无障碍访问支持，添加了ARIA标签、角色和状态属性，改进屏幕阅读器兼容性。
  * 新增可访问性演示示例和完整的多语言本地化支持。

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yangyanAurora committed:
feat: 添加部分组件无障碍功能支持
- button
- Checkbox
- Countdown
- Empty
- Image
- InputNumber
- Price
- Progress
- NavBar
- NoticeBar
- Radio
- Switch
- Tabbar
- Tabs
- Toast

yangyanAurora committed:
Update checkbox.tsx

yangyanAurora committed:
Update src/packages/countdown/demos/h5/demo10.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

yangyanAurora committed:
Update demo.taro.tsx

yangyanAurora committed:
Update demo.tsx

yangyanAurora committed:
Update src/packages/noticebar/demos/h5/demo12.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

yangyanAurora committed:
fix: update

yangyanAurora committed:
Update src/packages/price/price.taro.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

yangyanAurora committed:
Update price.taro.tsx

yangyanAurora committed:
Update price.tsx

yangyanAurora committed:
Update src/packages/radio/radio.taro.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

yangyanAurora committed:
Update src/packages/radio/radio.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

yangyanAurora committed:
fix: update

yangyanAurora committed:
fix: update

yangyanAurora committed:
Merge remote-tracking branch 'upstream/feat_v3.x_cpp' into feat_v3.x

yangyanAurora committed:
feat: 添加组件无障碍支持
- Dialog

yangyanAurora committed:
fix: update

xiaoyatong commented on src/packages/button/button.taro.tsx:
```diff
@@ -161,6 +162,8 @@ export const Button = React.forwardRef<HTMLButtonElement, Partial<ButtonProps>>(
         style={{ ...getStyle, ...style }}
         onClick={(e) => handleClick(e as any)}
+        ariaRole={role}
+        ariaDisabled={disabled}
```
加一个 button.tsx 下的多语言

xiaoyatong commented on src/packages/dialog/content.tsx:
```diff
@@ -58,6 +60,9 @@ export const Content: FunctionComponent<
       style={style}
       onClick={handleClick}
+      role={ariaRole}
+      aria-label={ariaLabel}
```
这个是否可以生效~看上去不符合react的语法~~

xiaoyatong commented on src/packages/dialog/dialog.taro.tsx:
```diff
@@ -247,7 +250,13 @@ export const BaseDialog: FunctionComponent<Partial<TaroDialogProps>> & {
+        ariaRole="button"
+        ariaLabel={locale.close}
+        tabindex={0}
+      >
```
icon库升级到 @nutui/icons-react-taro 3.0.2-cpp.3.beta.3

xiaoyatong commented on src/packages/dialog/dialog.taro.tsx:
```diff
@@ -305,6 +316,7 @@ export const BaseDialog: FunctionComponent<Partial<TaroDialogProps>> & {
           closeOnOverlayClick={closeOnOverlayClick}
           lockScroll={lockScroll}
           onClick={onHandleClickOverlay}
+          ariaLabel="背景蒙层"
```
这里可以不要，写在overlay 组件里就行了

xiaoyatong commented on src/packages/dialog/dialog.taro.tsx:
```diff
@@ -283,6 +292,8 @@ export const BaseDialog: FunctionComponent<Partial<TaroDialogProps>> & {
         footerDirection={footerDirection}
         visible={visible}
+        ariaRole={role}
+        ariaLabel={ariaLabel}
```
确认是不是会干扰到里边的内容读取

xiaoyatong commented on src/packages/inputnumber/inputnumber.taro.tsx:
```diff
@@ -58,6 +61,16 @@ export const InputNumber: FunctionComponent<
+      plus: '加',
+    },
+    'en-US': {
+      mins: 'Minus',
```
minus

xiaoyatong commented on src/packages/navbar/navbar.taro.tsx:
```diff
@@ -67,6 +67,7 @@ export const NavBar: FunctionComponent<Partial<TaroNavBarProps>> = (props) => {
               [`${classPrefix}-left-back-children-rtl`]: left && rtl,
             })}
             onClick={(e) => onBackClick(e)}
+            ariaLabel="back"
```
暴露一个属性、用户可以自定义

xiaoyatong replied:
类似close

xiaoyatong commented on src/packages/noticebar/noticebar.taro.tsx:
```diff
@@ -489,6 +491,7 @@ export const NoticeBar: FunctionComponent<
             onClick={handleClickIcon}
+            ariaLabel={`${closeable ? 'close' : rightIconAriaLabel}`}
           >
             {rightIcon || <Close size={12} />}
```
可以写在496 行的close上~

xiaoyatong commented on src/packages/overlay/overlay.taro.tsx:
```diff
@@ -15,6 +15,7 @@ export const defaultOverlayProps: TaroOverlayProps = {
   onClick: () => {},
   afterShow: () => {},
   afterClose: () => {},
+  ariaLabel: '蒙层',
```
类似close

xiaoyatong commented on src/packages/popup/popup.taro.tsx:
```diff
@@ -124,6 +126,7 @@ export const Popup: FunctionComponent<
   )
   const [popupHeight, setPopupHeight] = useState('')
   const [topBottom, setTopBottom] = useState('')
+  const role = 'dialog'
```
popup

xiaoyatong replied:
优先修改 popup

xiaoyatong commented on src/packages/switch/switch.taro.tsx:
```diff
@@ -29,6 +29,7 @@ export const Switch: FunctionComponent<Partial<TaroSwitchProps>> = (props) => {
     onChange,
     loading: propLoading,
     onLoadingChange,
+    ariaLabel,
```
默认可以加一下 切换按钮+关闭

yangyanAurora committed:
fix: merge branch 'upstream feat_v3.x_cpp ' to 'feat_v3.x'

yangyanAurora committed:
fix: update merge issue

yangyanAurora committed:
fix: update about ARIA including component:
- Image
- Countdown
- InputNumber
- NavBar
- NoticeBar
- Price
- Progress
- Tabbar
- Tabs
